### PR TITLE
refactor: updated the npm packages. refactored graphql and redis. fixed linting issues

### PR DESCRIPTION
**Pull request by IgnatG**
### Introduction:
A number of issues have been reported by SonarCloud, including linting errors and potential bugs

### Resolution
- removed Graphql from Portal-API. This was supposed to be decommissioned before going live in April 2022, but it was always postponed. The removal of GraphQL helped removing all security vulnerabilities from Portal as well
- updated the npm packages
- fixed linting issues
- updated docker-compose files to include Redis and Node_ENV env variables

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -40,6 +40,7 @@ services:
       - MONGODB_URI=mongodb://root:r00t@dtfs-submissions-data:27017/?authMechanism=DEFAULT
       - DTFS_CENTRAL_API=http://localhost:5005
       - TZ=Europe/London
+      - NODE_ENV=development
```
Comment (abhi-markan) [inline docker-compose.yml:43]: Can this and all of the below be picked up from env please `${{ env.NODE_END }}`

File: trade-finance-manager-ui/server/healthcheck.js [hunk 2]
```diff
@@ -3,20 +3,17 @@ const express = require('express');
...
-    },
+    message: 'OK',
+    timestamp: Date.now(),
   };
-
   try {
     res.status(200).send(data);
-  } catch (e) {
-    data.message = e;
+  } catch (error) {
```
Comment (abhi-markan) [inline trade-finance-manager-ui/server/healthcheck.js:15]: Can we update other catch blocks as well where `e` is `error` please?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Looks good, can you please add in description why `GQL` has been removed from `portal` please?

File: docker-compose.yml (hunk 1 above)
Comment (IgnatG) [reply docker-compose.yml]: good shout - this is updated now

File: trade-finance-manager-ui/server/healthcheck.js (hunk 2 above)
Comment (IgnatG) [reply trade-finance-manager-ui/server/healthcheck.js:15]: updated now

File: .github/workflows/pr_e2e_tests.yml [hunk 3]
```diff
@@ -49,6 +49,9 @@ env:
   UKEF_TFM_API_SYSTEM_KEY: ${{ secrets.UKEF_TFM_API_SYSTEM_KEY }}
   UKEF_TFM_API_REPORTS_KEY: ${{ secrets.UKEF_TFM_API_REPORTS_KEY }}
   ESTORE_URL: ${{ secrets.ESTORE_URL }}
+  REDIS_HOSTNAME: ${{ secrets.REDIS_HOSTNAME }}
+  REDIS_PORT: ${{ secrets.REDIS_PORT }}
+  NODE_END: ${{ secrets.NODE_ENV }}
```
Comment (abhi-markan) [inline .github/workflows/pr_e2e_tests.yml:54]: Typo `NODE_ENV` please.

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Good work, one minor comment.
Should we update node to `v20`?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Good work, one minor comment.
Should we update node to `v20`?

**Comment by IgnatG** [comment]
> Good work, one minor comment.
> Should we update node to `v20`?

Thanks @abhi-markan . It should have v20 already. We did that a few days ago

**Pull request merged**